src/observability/publisher.go

```diff
@@ -41,7 +50,51 @@ func ProcessSystemSummary() {
 			// clear each published entry from data map
 			delete(tempSummarizerMap, ss)
 		}
+		log.Info().Msgf("Published %v events in %v", count, time.Since(timeVar))
 		PublisherMutex.Unlock()
 	}
 	ProcessSystemSummaryWg.Wait()
 }
+
+func getAggregatedSummaryMap(tempSummarizerMap map[types.SystemSummary]types.SysSummaryTimeCount) map[types.SystemSummary]types.SysSummaryTimeCount {
+
+	var fileArr []string
+	log.Info().Msgf("Events before aggregation: [%v]", len(tempSummarizerMap))
+
+	fileSumMap := make(map[types.SystemSummary]types.SysSummaryTimeCount)
+	updatedSummarizerMap := make(map[types.SystemSummary]types.SysSummaryTimeCount)
+	for tss, tsstc := range tempSummarizerMap {
+		if tss.Operation == types.FileOperation {
+			fileArr = append(fileArr, tss.Destination)
+			fileSumMap[tss] = tsstc
+		} else {
+			updatedSummarizerMap[tss] = tsstc
+		}
+	}
+	tempSummarizerMap = updatedSummarizerMap
+
+	aggregatedFilePaths := common.AggregatePaths(fileArr)
+
+	log.Info().Msgf("Aggregated file paths: [%v]", len(aggregatedFilePaths))
+	for ss, sstc := range fileSumMap {
+		for _, path := range aggregatedFilePaths {
+			if strings.HasPrefix(ss.Destination, path.Path) && (len(ss.Destination) == len(path.Path) || ss.Destination[len(strings.TrimSuffix(path.Path, "/"))] == '/') {
+				ss.Destination = path.Path
+				break
+			}
+		}
```

Comment:
Since the iteration is happening only via one loop do we need to clear elements from the map? I thought adding this delete() might increase the time complexity of the loop. What do you suggest @seswarrajan ?